File: epi_judge_java/epi/ReverseDigits.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

public class ReverseDigits {

    @EpiTest(testDataFile = "reverse_digits.tsv")
    public static long reverse(int x) {
        int sign = (x < 0 ? -1 : 1);
        return sign * Long.parseLong(new StringBuilder().append(Math.abs((long) x)).reverse().toString());
    }

    public static void main(String[] args) {
        System.exit(
                GenericTest
                        .runFromAnnotations(args, "ReverseDigits.java",
                                new Object() {
                                }.getClass().getEnclosingClass())
                        .ordinal());
    }
}

File: epi_judge_python/reverse_digits.py
from test_framework import generic_test


def reverse(x: int) -> int:
    is_negative = x < 0
    x = abs(x)

    result = 0

    while x:
        result *= 10
        result += x % 10
        x //= 10

    return -result if is_negative else result


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main('reverse_digits.py',
                                       'reverse_digits.tsv', reverse))
